### PR TITLE
fix(extensions): consolidate extensions headers returned to UI by ext…

### PR DESCRIPTION
**Pull request by peusebiu**
…ensions

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/common/common.go [hunk 1]
```diff
@@ -283,3 +285,28 @@ func GetManifestArtifactType(manifestContent ispec.Manifest) string {
 
 	return manifestContent.Config.MediaType
 }
+
+func AddExtensionSecurityHeaders(h http.Handler) http.HandlerFunc { //nolint:varnamelen
```
Comment (alexstan12) [inline pkg/common/common.go:289]: Could this function also return a mux.MiddlewareFunc to be applied on the router (mgmtRouter for eg)? does it make sense to be treated as a mdw as well ?

File: pkg/extensions/extension_mgmt.go [hunk 2]
```diff
@@ -82,20 +82,16 @@ func (mgmt *mgmt) handler() http.Handler {
...
 	if config.Extensions.Mgmt != nil && *config.Extensions.Mgmt.Enable {
 		log.Info().Msg("setting up mgmt routes")
 
 		mgmt := mgmt{config: config, log: log}
 
-		router.PathPrefix(constants.ExtMgmt).Methods("GET").Handler(addMgmtSecurityHeaders(mgmt.handler()))
+		allowedMethods := zcommon.AllowedMethods(http.MethodGet)
+
+		mgmtRouter := router.PathPrefix(constants.ExtMgmt).Subrouter()
+		mgmtRouter.Use(zcommon.ACHeadersHandler(allowedMethods...))
```
Comment (alexstan12) [inline pkg/extensions/extension_mgmt.go:94]: see the above comment on AddExtensionSecurityHeaders .

**Review by alexstan12 — DISMISSED** [review]
LGTM

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**